examples/nidcpower_source_dc_voltage/measurement.py

```diff
@@ -68,96 +66,91 @@ def measure(
     """Source and measure a DC voltage with an NI SMU."""
     logging.info("Executing measurement: pin_names=%s voltage_level=%g", pin_names, voltage_level)
 
-    session_management_client = nims.session_management.Client(
-        grpc_channel=measurement_service.get_channel(
-            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
-            service_class=nims.session_management.GRPC_SERVICE_CLASS,
-        )
+    session_management_client = create_session_management_client(measurement_service)
+
+    reservation = reserve_session(
```

Comment:
This leaks the reservation, making the next measurement hang forever. Use `with reserve_session(...) as reservation:` to manage the lifetime of the reservation.

Reply:
Enclosed reservation code inside `with`.